test/e2e/storage/volume_expand.go

```diff
@@ -0,0 +1,196 @@
+		}()
+
+		By("Waiting for file system resize to finish")
+		pvc, err = waitForFSResize(pvc, c)
+		Expect(err).NotTo(HaveOccurred(), "while waiting for fs resize to finish")
+
+		pvcConditions := pvc.Status.Conditions
+		Expect(len(pvcConditions)).To(Equal(0), "pvc should not have conditions")
+	})
+})
+
+func createResizableStorageClass(t storageClassTest, ns string, suffix string, c clientset.Interface) (*storage.StorageClass, error) {
+	stKlass := newStorageClass(t, ns, suffix)
+	allowExpansion := true
+	stKlass.AllowVolumeExpansion = &allowExpansion
+
+	var err error
+	stKlass, err = c.StorageV1().StorageClasses().Create(stKlass)
+	return stKlass, err
+}
+
+func expandPVCSize(origPVC *v1.PersistentVolumeClaim, size resource.Quantity, c clientset.Interface) (*v1.PersistentVolumeClaim, error) {
+	pvcName := origPVC.Name
+	updatedPVC := origPVC.DeepCopy()
+
+	waitErr := wait.PollImmediate(2*time.Second, 30*time.Second, func() (bool, error) {
+		var err error
+		updatedPVC, err = c.CoreV1().PersistentVolumeClaims(origPVC.Namespace).Get(pvcName, metav1.GetOptions{})
+		if err != nil {
+			return false, fmt.Errorf("Error fetching pvc %s for resizing with %v", pvcName, err)
+		}
+
+		updatedPVC.Spec.Resources.Requests[v1.ResourceStorage] = size
+		updatedPVC, err = c.CoreV1().PersistentVolumeClaims(origPVC.Namespace).Update(updatedPVC)
+		if err == nil {
+			return true, nil
+		}
+		framework.Logf("Error updating pvc %s with %v", pvcName, err)
+		return false, nil
+	})
+	return updatedPVC, waitErr
+}
+
+func waitForControllerVolumeResize(pvc *v1.PersistentVolumeClaim, c clientset.Interface) error {
+	pvName := pvc.Spec.VolumeName
+	return wait.PollImmediate(2*time.Second, 20*time.Minute, func() (bool, error) {
+		pvcSize := pvc.Spec.Resources.Requests[v1.ResourceStorage]
+
+		pv, err := c.CoreV1().PersistentVolumes().Get(pvName, metav1.GetOptions{})
+		if err != nil {
+			return false, fmt.Errorf("Error fetching pv %s for resizing %v", pvName, err)
+		}
+
+		pvSize := pv.Spec.Capacity[v1.ResourceStorage]
+
+		// If pv size is greater or equal to requested size that means controller resize is finished.
+		if pvSize.Cmp(pvcSize) >= 0 {
+			return true, nil
+		}
+		return false, nil
+	})
+}
+
+func waitForFSResize(pvc *v1.PersistentVolumeClaim, c clientset.Interface) (*v1.PersistentVolumeClaim, error) {
+	var updatedPVC *v1.PersistentVolumeClaim
+	waitErr := wait.PollImmediate(2*time.Second, 20*time.Minute, func() (bool, error) {
+		var err error
+		updatedPVC, err = c.CoreV1().PersistentVolumeClaims(pvc.Namespace).Get(pvc.Name, metav1.GetOptions{})
+
+		if err != nil {
+			return false, fmt.Errorf("Error fetching pvc %s for checking for resize status : %v", pvc.Name, err)
+		}
+
+		pvcSize := updatedPVC.Spec.Resources.Requests[v1.ResourceStorage]
+		pvcStatusSize := updatedPVC.Status.Capacity[v1.ResourceStorage]
+
+		// If pvc's status field is of more size than pvc size then we are done here
```

Comment:
reword to something like:
_// If pvc's status field size is greater than pvc's size then done_

Reply:
ack